invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/controls/PublishButton/SubmitReviewOrPublishButton.js

```diff
@@ -36,7 +36,7 @@ class SubmitReviewOrPublishComponent extends Component {
       isDOIRequired !== undefined && // isDOIRequired is undefined when no value was provided from Invenio-app-rdm
       !isDOIRequired &&
       noINeedDOI &&
-      Object.keys(formik?.values?.pids).length === 0;
+      Object.keys(formik?.values?.pids?.doi || {}).length === 0;
```

Comment:
I think it would be nice to put explicit comment on why (mention other pids, OAI etc)